.gitignore

```diff
@@ -1,4 +1,5 @@
 /target
+/server/target
```

Comment:
While cargo workspace has a common target directory in `/target`, we still need the project-specific directories because sqlx adds some files 🤷‍♂️